numojo/math/math_funcs.mojo

```diff
@@ -192,16 +194,16 @@ struct Vectorized(Backend):
 
         @parameter
         fn closure[simdwidth: Int](i: Int):
-            var simd_data1 = array1.load[width=opt_nelts](i)
-            var simd_data2 = array2.load[width=opt_nelts](i)
-            result_array.store[width=opt_nelts](
-                i, func[dtype, opt_nelts](simd_data1, simd_data2)
+            var simd_data1 = array1.load[width=simdwidth](i)
+            var simd_data2 = array2.load[width=simdwidth](i)
+            result_array.store[width=simdwidth](
+                i, func[dtype, simdwidth](simd_data1, simd_data2)
             )
 
         vectorize[closure, opt_nelts](result_array.num_elements())
         return result_array
 
-    fn _math_func_compare_2_tensors[
+    fn math_func_compare_2_tensors[
```

Comment:
Shall we rename this as `SIMD` instead of `tensors`?

Reply:
It would be `array` or `ndarray` as the function takes two arrays and compares them using the function passed as an argument. Tensor is technically accurate just not the name of the struct that we actually use.

We can change it, but it is in a lot of places.